packages/components/forms/src/Multiselect/Multiselect.tsx

```diff
@@ -0,0 +1,286 @@
+import React, { useRef, useState } from 'react';
+import { cx } from 'emotion';
+
+import { mergeRefs, type CommonProps } from '@contentful/f36-core';
+import { Button, IconButton } from '@contentful/f36-button';
+import { TextInput, type TextInputProps } from '../TextInput';
+import { CloseIcon, ChevronDownIcon, SearchIcon } from '@contentful/f36-icons';
+import { SkeletonContainer, SkeletonBodyText } from '@contentful/f36-skeleton';
+import { Popover } from '@contentful/f36-popover';
+import { Subheading } from '@contentful/f36-typography';
+
+import { getMultiselectStyles } from './Multiselect.styles';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface MultiselectProps
+  extends CommonProps,
+    Pick<
+      TextInputProps,
+      | 'isDisabled'
+      | 'isInvalid'
+      | 'isReadOnly'
+      | 'isRequired'
+      | 'id'
+      | 'defaultValue'
+    > {
+  /** Select Options */
+  children?: React.ReactNode;
+
+  /**
+   * Set a custom icon for the text input
+   */
+  startIcon?: React.ReactElement;
+
+  /**
+   * Placeholder shown before selecting any elements. Defaults to 'Select one or more items'
+   */
+  placeholder?: string;
+
+  /**
+   * current Selected items, to be shown on the trigger button
+   */
+  currentSelection?: Array<string>;
+
+  /**
+   * If this is set to `true` there will be a search input inside the drawer
+   * @default false
+   */
+  hasSearch?: boolean;
+
+  /**
+   * Function called whenever the search input value changes
+   */
+  onSearchValueChange?: (event: React.ChangeEvent) => void;
+
+  /**
+   * This is the value will be passed to the `placeholder` prop of the input.
+   * @default "Search"
+   */
+  searchPlaceholder?: string;
+
+  /**
+   * A message that will be shown when it is not possible to find any option that matches the input value
+   * @default "No matches"
+   */
+  noMatchesMessage?: string;
+
+  /**
+   * Use this prop to get a ref to the input element of the component
+   */
+  searchInputRef?: React.Ref<HTMLInputElement>;
+  /**
+   * Use this prop to get a ref to the toggle button of the component
+   */
+  toggleRef?: React.Ref<HTMLButtonElement>;
+  /**
+   * Use this prop to get a ref to the list of items of the component
+   */
+  listRef?: React.Ref<HTMLUListElement>;
+
+  /**
+   * It sets the width of the list
+   * @default "auto"
+   */
+  listWidth?: 'auto' | 'full';
```

Comment:
Elsewhere we use a `isFullWidth` boolean, seems like the same could apply here, or do we expect to extend this union at some point?

Based on the previous comment it could be combined into something like

```
listProps: {
  isFullWidth?: boolean;
  ref?: React.Ref<HTMLUListElement>;
}
```

Reply:
i m not sure how the proper syntax would be, is there a component where we use this approach with subcomponent properties?

Reply:
Datepicker for instance takes `inputProps`: https://f36.contentful.com/components/datepicker#visual-states